### PR TITLE
feature: add spec annotation flag

### PR DESCRIPTION
**Pull request by Ace-Tang**
spec annotation flag pass all user defined runtime annotation to
runtime, note that whether the annotation will take effect depend
on your runtime inplement.

Signed-off-by: Ace-Tang <aceapril@126.com>

### CODE REVIEW

**Comment by Ace-Tang** [comment]
Use this specify flag make the previous steps more simple.
1. do not need to parse defined label format.
2. do not need add more logic in hook.
3. since hook or something need to transfer label to annontation, and this annotation need to be add into spec in `setupAnnotations`, we can simplely add annotation passed by flag `spec-annotation` in `setupAnnotations`.

**Comment by yyb196** [comment]
LGTM

File: apis/types/container_config.go [hunk 1]
```diff
@@ -108,6 +108,9 @@ type ContainerConfig struct {
 
 	// The working directory for commands to run in.
 	WorkingDir string `json:"WorkingDir,omitempty"`
+
+	// Annotations send to runtime spec.
+	SpecAnnotation map[string]string `json:"SpecAnnotation,omitempty"`
```
Comment (allencloud) [inline apis/types/container_config.go]: Please add this in swagger.yml.

**Comment by Ace-Tang** [comment]
do we need show `SpecAnnotation` in pouch inspect.

**Comment by allencloud** [comment]
Please add some unit test for the function validateAnnotation and add integration test for the field.

File: apis/swagger.yml [hunk 2]
```diff
@@ -1751,6 +1751,11 @@ definitions:
         x-nullable: true
         additionalProperties:
           type: "string"
+      SpecAnnotation:
+        description: "nnotations send to runtime spec."
```
Comment (rudyfly) [inline apis/swagger.yml]: s/nnotations/annotations ?

**Comment by Ace-Tang** [comment]
@allencloud , has updated. About `validate*` functions test, I wanna add unit tests for these kind functions in other pr.

File: test/cli_start_test.go [hunk 3]
```diff
@@ -233,3 +233,14 @@ func (suite *PouchStartSuite) TestStartWithPrivilege(c *check.C) {
...
+
+// TestStartWithAnnotation starts a container with annotation.
+func (suite *PouchStartSuite) TestStartWithAnnotation(c *check.C) {
+	name := "start-annotation"
+
+	res := command.PouchRun("create", "--name", name, "--annotation", "a=b", busyboxImage)
+	res.Assert(c, icmd.Success)
+	defer DelContainerForceMultyTime(c, name)
+
+	command.PouchRun("start", name).Assert(c, icmd.Success)
```
Comment (allencloud) [inline test/cli_start_test.go]: I think after start, we need to check these annotation as well with `pouch inspect`.

**Comment by allencloud** [comment]
This pr will be related to pr https://github.com/alibaba/pouch/pull/1041. If this is merged first, https://github.com/alibaba/pouch/pull/1041 needs rebased. @HusterWan

**Comment by HusterWan** [comment]
got it

File: test/cli_start_test.go (hunk 3 above)
Comment (Ace-Tang) [reply test/cli_start_test.go]: this has been done in `test/cli_run_test.go`

Comment (allencloud) [reply test/cli_start_test.go]: No, I do not think they are the same. They have called the same API, but in cli side, they called different code. Double check would be nicer to guarantee the code quality.

Comment (Ace-Tang) [reply test/cli_start_test.go]: Follow the previous logic, inspect test in doing in cli_run_test.go, not in start test, as I agree with you, this also related all test case in `test/cli_start_test.go`.

Comment (Ace-Tang) [reply test/cli_start_test.go]: oh! I miss create test, this things should be done there too.

**Comment by allencloud** [comment]
LGTM, too.

**Pull request merged**